### PR TITLE
[Image Thumbnails] Refactor auto-format thumbnail configs generation for reusability

### LINKED ISSUE #11508: [ImageThumbnails] Respect quality configuration for auto formats in pimcore:thumbnails:image command
### Expected behavior

please see https://github.com/pimcore/pimcore/commit/77bcc93b019ef83412f829948ca6fd279bc0dd59#commitcomment-67228922

### Actual behavior

please see https://github.com/pimcore/pimcore/commit/77bcc93b019ef83412f829948ca6fd279bc0dd59#commitcomment-67228922

### Steps to reproduce

1. Set quality of image thumbnail auto format as mentioned here https://pimcore.com/docs/pimcore/current/Development_Documentation/Assets/Working_with_Thumbnails/Image_Thumbnails.html#page_Customize-Auto-Web-Optimized-Format
2. Run command to generate image thumbnail `bin/console pimcore:thumbnails:image --thumbnails=MyThumbnail`
3. Image thumbnails are generated with quality defined in thumbnail config not the custom config in YAML.

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #11508

### CODE REVIEW

File: models/Asset/Image/Thumbnail/Config.php [hunk 1]
```diff
@@ -930,4 +930,34 @@ public function __clone()
             }
         }
     }
+
+    /**
+     * @return array
+     */
+    public static function getAutFormats(): array
```
Comment (brusch) [inline models/Asset/Image/Thumbnail/Config.php]:
```suggestion
    public static function getAutoFormats(): array
```
Typo 😉

File: models/Asset/Image/Thumbnail/Config.php [hunk 2]
```diff
@@ -930,4 +930,34 @@ public function __clone()
...
+    {
+        return \Pimcore::getContainer()->getParameter('pimcore.config')['assets']['image']['thumbnails']['auto_formats'];
+    }
+
+    /**
+     * @return Config[]
+     */
+    public function getAutoFormatThumbnailConfigs(): array
+    {
+        $autFormatThumbnails = [];
```
Comment (brusch) [inline models/Asset/Image/Thumbnail/Config.php]:
```suggestion
        $autoFormatThumbnails = [];
```
another typo, maybe double check your `o` key 😂

File: models/Asset/Image/Thumbnail.php [hunk 3]
```diff
@@ -308,19 +308,10 @@ public function getHtml($options = [])
...
-                                if (!empty($autoFormatConfig['quality'])) {
-                                    $thumbConfigAutoFormat->setQuality($autoFormatConfig['quality']);
-                                }
-
-                                $sourceWebP = $this->getSourceTagHtml($thumbConfigAutoFormat, $mediaQuery, $image, $options);
-                                if (!empty($sourceWebP)) {
-                                    $html .= "\t" . $sourceWebP . "\n";
-                                }
+                        foreach ($thumbConfig->getAutoFormatThumbnailConfigs() as $autoFormatConfig) {
+                            $sourceWebP = $this->getSourceTagHtml($autoFormatConfig, $mediaQuery, $image, $options);
```
Comment (brusch) [inline models/Asset/Image/Thumbnail.php]: Maybe we should use a more generic variable name here instead of `$sourceWebP`.

**Review by brusch — CHANGES_REQUESTED** [review]

File: models/Asset/Image/Thumbnail/Config.php (hunk 2 above)
Comment (dvesh3) [reply models/Asset/Image/Thumbnail/Config.php]: Done...yes guess time to change my old keyboard.

File: models/Asset/Image/Thumbnail.php (hunk 3 above)
Comment (dvesh3) [reply models/Asset/Image/Thumbnail.php]: Changed it to `$autoFormatThumbnailHtml`

File: models/Asset/Image/Thumbnail/Config.php [hunk 4]
```diff
@@ -930,4 +930,34 @@ public function __clone()
...
+     * @return array
+     */
+    public static function getAutoFormats(): array
+    {
+        return \Pimcore::getContainer()->getParameter('pimcore.config')['assets']['image']['thumbnails']['auto_formats'];
+    }
+
+    /**
+     * @return Config[]
+     */
```
Comment (brusch) [inline models/Asset/Image/Thumbnail/Config.php]:
```suggestion
    /**
     * @internal
     * @return Config[]
     */
```

File: models/Asset/Image/Thumbnail/Config.php [hunk 5]
```diff
@@ -930,4 +930,34 @@ public function __clone()
             }
         }
     }
+
+    /**
+     * @return array
+     */
```
Comment (brusch) [inline models/Asset/Image/Thumbnail/Config.php]:
```suggestion
    /**
     * @internal
     * @return array
     */
```

**Review by brusch — CHANGES_REQUESTED** [review]

**Pull request merged**